### DISCUSSION TITLE
Does juicefs support reverse proxy for multiple minio server?

### DISCUSSION THREAD

**Discussion by mdianjun** [juicedata/juicefs#861] [General]
Does juicefs have load balance function like https://github.com/minio/sidekick ，which is able to proxy multiple minio server instances?

**Comment by mdianjun** [comment]
I make a test for this case.
I have 4 minio servers:  192.168.1.21{0..3}, the format command is:
```
juicefs format --storage minio \
    --bucket "http://192.168.1.21%d:9999/bkt1" \
    --shards 4 \
   ...
```

Juicefs splits the file into a few objects, and then distributes objects across all shards. If one of the minio servers is down, the progress of uploading objects to the failed minio server will hang. Juicefs have **no health checking** about the minio servers, and skip the failed one.

**Reply by xiaogaozi** [reply]
Hi @mdianjun, as you mentioned, it's the expected behavior. JuiceFS client relies on **the object storage** itself to guarantee availability.

**Reply by davies** [reply]
@mdianjun When multiple buckets are used as objects store, the objects are distributed based on hash of key, so we can't change which bucket to use, otherwise we can't find the object back.

**Reply by jkiebzak** [reply]
@davies does this also mean we cannot add/remove buckets to a file system once we format it?

**Reply by davies** [reply]
Yes, unfortunately.

**Comment by davies** [comment]
You can setup sidekick or other load balancer by yourself, or maybe we can have a failover policy to pick another one as backup (migrate them back when primary one is healthy).
